aggie-coding-club/Rev-Registration
# Associate user with session

## Description

This adds the functionality for user's information to be saved between sessions and computers if they are logged in. Before, all their session data would be destroyed upon logout. Now, it isn't. This is done by saving it in a different session that is never destroyed (the data session) which can then be accessed via `retrieve_data_session` when necessary. If a user is not logged in, `retrieve_data_session` will return the usual `request.session`.

## Rationale


## How to test

I wrote some tests for `retireve_data_session` so running those and checking if they pass would be a good place to start. Make sure to migrate the database because this PR has a new model that links a user to their data session. Aside from the manual tests, I would just try calling all the APIs we have and making sure they don't return forbidden or not work or something. I would also save a schedule, logout, and then log back in and see if it's still there since that was the main point of adding this in the first place.

## Related tasks

Closes # I don't think this had an issue actually. Should I make one? Maybe #267 but that's kind of a stretch and I think Ryan already did that one.
